cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -604,3 +604,56 @@ func (s *Server) UpdateInstalledPackage(ctx context.Context, request *corev1.Upd
 		InstalledPackageRef: updatedRef,
 	}, nil
 }
+
+// DeleteInstalledPackage Deletes an installed package managed by the 'kapp_controller' plugin
+func (s *Server) DeleteInstalledPackage(ctx context.Context, request *corev1.DeleteInstalledPackageRequest) (*corev1.DeleteInstalledPackageResponse, error) {
+	log.Infof("+kapp-controller DeleteInstalledPackage")
+
+	// Validate the request
+	if request == nil || request.GetInstalledPackageRef() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no request InstalledPackageRef provided")
+	}
+	if request.GetInstalledPackageRef().GetContext().GetNamespace() == "" || request.GetInstalledPackageRef().GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "required context or identifier not provided")
+	}
+
+	// Retrieve the proper parameters from the request
+	packageRef := request.GetInstalledPackageRef()
+	identifier := request.GetInstalledPackageRef().GetIdentifier()
+	namespace := packageRef.GetContext().GetNamespace()
+	cluster := packageRef.GetContext().GetCluster()
+
+	if cluster == "" {
+		cluster = s.globalPackagingCluster
+	}
+
+	typedClient, _, err := s.GetClients(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	pkgInstall, err := s.getPkgInstall(ctx, cluster, namespace, identifier)
+	if err != nil {
+		return nil, errorByStatus("get", "PackageInstall", identifier, err)
+	}
+
+	// Delete all the associated secrets
```

Comment:
Again, I'd do this after deleting the package (and longer-term, only if we created the associated secret.)